File: android/src/main/java/co/alephnull/reactnative/silicon/verifier/VerifierTypes.kt
package co.alephnull.reactnative.silicon.verifier

import co.alephnull.reactnative.silicon.PayloadType
import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record
import expo.modules.kotlin.types.Enumerable

enum class VerifyAlgorithm(val value: String) : Enumerable {
    ES256("ES256"),
    ES384("ES384"),
    ES512("ES512"),
    RS256("ES256"),
    RS384("ES384"),
    RS512("ES512"),
    PS256("ES256"),
    PS384("ES384"),
    PS512("ES512"),
}

class VerifyOptions : Record {
    @Field var alias: String? = null
    @Field var pubkey: PayloadType? = null
    @Field var algorithm: VerifyAlgorithm? = null
}

class BridgeVerifyOptions : Record {
    @Field var alias: String? = null
    @Field var pubkeyStr: String? = null // Base64 encoded public key
    @Field var pubkeyBytes: ByteArray? = null // Raw Bytes
    @Field var algorithm: VerifyAlgorithm? = null
}

File: ios/Verifier/VerifierTypes.swift
import ExpoModulesCore

enum VerifyAlgorithms: String, Enumerable {
    case ES256 = "ES256"
    case ES384 = "ES384"
    case ES512 = "ES512"
    case RS256 = "RS256"
    case RS384 = "RS384"
    case RS512 = "RS512"
    case PS256 = "PS256"
    case PS384 = "PS384"
    case PS512 = "PS512"
}

struct VerifyOptions {
    var alias: String? = nil
    var pubkey: PayloadType? = nil
    var algorithm: VerifyAlgorithms?
}

struct BridgeVerifyOptions: Record {
    @Field
    var alias: String? = nil
    
    @Field
    var pubkeyStr: String? = nil
    
    @Field
    var pubkeyBytes: Data? = nil
    
    @Field
    var algorithm: VerifyAlgorithms?
}
